src/conversion/viewconversiondispatcher.js

```diff
@@ -128,14 +134,31 @@ export default class ViewConversionDispatcher {
 	 * viewItem Part of the view to be converted.
 	 * @param {Object} [additionalData] Additional data to be passed in `data` argument when firing `ViewConversionDispatcher`
 	 * events. See also {@link ~ViewConversionDispatcher#event:element element event}.
-	 * @returns {module:engine/model/documentfragment~DocumentFragment} Model document fragment that is a result of the conversion process.
+	 * @returns {module:engine/model/documentfragment~DocumentFragment} Model data that is a result of the conversion process
+	 * wrapped by DocumentFragment.
 	 */
 	convert( viewItem, additionalData = {} ) {
 		this.fire( 'viewCleanup', viewItem );
 
 		const consumable = ViewConsumable.createFrom( viewItem );
+		const conversionResult = this._convertItem( viewItem, consumable, additionalData );
 
-		return this._convertItem( viewItem, consumable, additionalData );
+		// When conversion output is not a Node or Element we just return it.
+		if ( !( conversionResult instanceof ModelNode || conversionResult instanceof ModelDocumentFragment ) ) {
+			return conversionResult;
+		}
```

Comment:
This is because conversion output doesn't have to be a node. Theoretically it can be [any value](https://github.com/ckeditor/ckeditor5-engine/blob/master/tests/conversion/viewconversiondispatcher.js#L75) but we use it to provide converted element. @scofalik know the details :) I know that it might be confusing especially when there is no explanation in docs.